script.skipit/resources/lib/skipit.py

```diff
@@ -67,23 +67,54 @@ def onAction(self, action):
           reset()
           skip_backward()
 
+
+def do_skip(skip_secs):
+    # We need to track the actual skip, in case we run past the end
+    totalTime = xbmc.Player().getTotalTime()
+    preSeekTime = xbmc.Player().getTime()
+
+    seek_to = preSeekTime + skip_secs;
+    if seek_to < 1 :
+        seek_to = 1;
+    if seek_to > (totalTime - 1):
+        seek_to = totalTime - 1;
+    xbmc.Player().seekTime(seek_to)
+    # Now work out how far we went (might have run to the end or start
+    postSeekTime = xbmc.Player().getTime()
+    return round(postSeekTime - preSeekTime)
+
 def skip_forward():
+
+    if not(xbmc.Player().isPlaying()):
+        close_me()
+        return
     # 
     # Skip forward the required amount
+    global addonname
     global skip_secs
     global going_forward
     going_forward = True
 
     reset_timer()
     halve()
     set_label(skip_secs, True)
-    command = build_command(skip_secs - slippage, True)
-    xbmc.executebuiltin(command)
+    requested_skip = skip_secs - slippage
+    actual_skip = do_skip(requested_skip)
+    if actual_skip != requested_skip:
+        skip_secs = abs(actual_skip)
+        command = 'Notification({},{})'
```

Comment:
Why not use  ` xbmcgui.Dialog().notification(heading, message[, icon, time, sound])`: https://codedocs.xyz/MartijnKaijser/xbmc/group__python___dialog.html#ga27157f98dddb21b44cc6456137977aa2

Reply:
I would prefer that. Previously I couldn't fathom the api for that one. Thanks